src/lib/viewers/ResultsViewer.ts

```diff
@@ -129,8 +134,71 @@ export class ResultsDetailViewer extends AbstractResultsViewer {
 	}
 }
 
+type ResultRow = {
+	id: number;
+	location: string;
+	name: string;
+	severity: string;
+}
+
+const TABLE_COLUMNS: Ux.Table.Columns<ResultRow> = {
+	id: {
+		header: getMessage(BundleName.ResultsViewer, 'summary.table.id-column'),
+	},
+	location: {
+		header: getMessage(BundleName.ResultsViewer, 'summary.table.location-column')
+	},
+	name: {
+		header: getMessage(BundleName.ResultsViewer, 'summary.table.name-column')
+	},
+	severity: {
+		header: getMessage(BundleName.ResultsViewer, 'summary.table.severity-column')
+	}
+};
+
 export class ResultsTableViewer extends AbstractResultsViewer {
-	public view(results: RunResults) {
-		throw new Error(`TODO: Table-formatted output is not available yet, but results were ${JSON.stringify(results)}`);
+	protected _view(results: RunResults) {
+		const resultRows: ResultRow[] = this.sortViolations(results.getViolations())
+			.map((v, idx) => {
+				const severity = v.getRule().getSeverityLevel();
+				const primaryLocation = v.getCodeLocations()[v.getPrimaryLocationIndex()];
+				return {
+					id: idx + 1,
+					location: `${primaryLocation.getFile()}:${primaryLocation.getStartLine()}:${primaryLocation.getStartColumn()}`,
+					name: v.getRule().getName(),
+					severity: `${severity.valueOf()} (${SeverityLevel[severity]})`
+				}
+			});
+		this.display.displayTable(resultRows, TABLE_COLUMNS);
+	}
+
+	private sortViolations(violations: Violation[]): Violation[] {
```

Comment:
It feels odd that each view sorts the violations differently. It results in the table view having different violation ids from the detail view. Should we have a single sort method that all views use?

I guess there could be a case for both... but I'd like to have the violation IDs the same in the views and in the files if at all possible.